### PR TITLE
fix: add what are signed messages link to page

### LINKED ISSUE #1430: [EIP-1271] Move link on the article to the top as on the design
## Bug description
 Move the link on the article to the top as on the design
## Environment
 - Browser: Chrome
 - Wallet: MetaMask
 - Chain: Goerli

## Steps to reproduce
 1. Go to the messages tab when there is at least one message
Current result: there is no possibility to read the article as the link on the article can be seen only on the empty Messages tab

## Expected result
Could we implement it according to the design?
## Screenshots
image

### PR DESCRIPTION
**Pull request by DiogoSoaress**
## What it solves

Resolves #1430 

## How this PR fixes it
Adds a link to the Messages page header if there are off-chain signed messages

## How to test it
1. Go to `Transactions > Messages` in a Safe with off-chain messages signed and you should see the link
2. Go to `Transactions > Messages` in a Safe without off-chain messages signed and you should not see the link above the table but instead in the `NoMessage` placeholder

## Screenshots
Screenshot 2022-12-27 at 12 17 02
Screenshot 2022-12-27 at 12 16 47

### CODE REVIEW

File: src/pages/transactions/messages.tsx [hunk 1]
```diff
@@ -5,6 +5,32 @@ import PaginatedMsgs from '@/components/safe-messages/PaginatedMsgs'
...
+      <SvgIcon component={InfoIcon} inheritViewBox color="border" fontSize="small" />
+      {/* TODO: Add link to help article */}
+      <ExternalLink suppressIcon href="#">
+        <Typography variant="body2" fontWeight={700}>
+          What are signed messages?
+        </Typography>
+      </ExternalLink>
+    </Box>
+  )
+}
```
Comment (katspaugh) [inline src/pages/transactions/messages.tsx:33]: This is perfectly extractable into its own file.

File: src/pages/transactions/messages.tsx [hunk 2]
```diff
@@ -5,6 +5,32 @@ import PaginatedMsgs from '@/components/safe-messages/PaginatedMsgs'
...
+
+  if (safeMessagesCount === 0) {
+    return null
+  }
+
+  return (
+    <Box display="flex" alignItems="center" gap={1}>
+      <SvgIcon component={InfoIcon} inheritViewBox color="border" fontSize="small" />
+      {/* TODO: Add link to help article */}
+      <ExternalLink suppressIcon href="#">
```
Comment (katspaugh) [inline src/pages/transactions/messages.tsx:26]: Could you do me a favor and replace all instances of `suppressIcon` with `noIcon`?

Comment (DiogoSoaress) [reply src/pages/transactions/messages.tsx:26]: sure! in a different PR I assume?

File: src/pages/transactions/messages.tsx (hunk 1 above)
Comment (DiogoSoaress) [reply src/pages/transactions/messages.tsx]: I've moved it to `src/components/transactions/SignedMessagesHelpLink/index.tsx`

File: src/components/transactions/SignedMessagesHelpLink/index.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import { useAppSelector } from '@/store'
+import { selectSafeMessages } from '@/store/safeMessagesSlice'
+
+const SignedMessagesHelpLink = () => {
+  const safeMessages = useAppSelector(selectSafeMessages)
+  const safeMessagesCount = safeMessages.data?.results.length ?? 0
+
+  if (safeMessagesCount === 0) {
+    return null
+  }
```
Comment (katspaugh) [inline src/components/transactions/SignedMessagesHelpLink/index.tsx:13]: Why hide it when no messages? The link is still relevant to the page.

File: src/pages/transactions/messages.tsx (hunk 2 above)
Comment (katspaugh) [reply src/pages/transactions/messages.tsx]: Right, to dev.

Comment (DiogoSoaress) [reply src/pages/transactions/messages.tsx]: https://github.com/safe-global/web-core/pull/1460

File: src/components/transactions/SignedMessagesHelpLink/index.tsx (hunk 3 above)
Comment (DiogoSoaress) [reply src/components/transactions/SignedMessagesHelpLink/index.tsx:13]: When we have no messages we have the link in the placeholder. The issue report also mentions it https://github.com/safe-global/web-core/issues/1430 in the `Steps to reproduce`

Comment (katspaugh) [reply src/components/transactions/SignedMessagesHelpLink/index.tsx:13]: Right, just saw it on your second screenshot. 👍

**Review by katspaugh — APPROVED** [review]

**Comment by katspaugh** [comment]
Feel free to merge to the epic branch, just update it from dev first and replace the noIcon prop.

**Comment by francovenica** [comment]
Looks good. 
Note that the link is still an empty anchor. Whenever we make an RC for this feature that should have to a have the real link

**Pull request merged**